sips/sip-021/sip-021-nakamoto.md

```diff
@@ -0,0 +1,541 @@
+- Stackers do not earn the native STX tokens for signing off on blocks. Instead, they receive PoX payouts and their stacked STX tokens eventually unlock. By contrast, stakers earn the native token by signing off on blocks.
+- Because anyone can mine STX, anyone can become a Stacker. There is no way for existing Stackers to "close ranks" and prevent someone from joining -- if Stackers refuse to sign a block with a stack-stx contract call, then a would-be Stacker would issue a stack-stx call via a Stacks-on-Bitcoin transaction. This forces all subsequent miners to produce blocks which materialize this stack-stx call, thereby forcing Stackers to choose between admitting the new Stacker or halting the chain forever. By contrast, there is no penalty for "closing ranks" on new stakers in PoS systems.
+
+# Activation
+
+## Determinations Prior to SIP Activation
+
+Some aspects of this proposal are best decided after a reference implementation approaches code completion. The following will be solidified at a later date but will strictly remain within the bounds described below.
+
+### Block Reward Distribution and MEV
+
+The Nakamoto system will use a variation of the Assumed Total Commitment with Carryforward (ATC-C) MEV mitigation strategy described in [this document][MEV-analysis-link] to allocate block rewards to miners. The probability a miner will win the sortition and be granted the current tenure will be based on a function that accounts for the total block commit spend on the blocks leading up to the current sortition.
+
+An example ATC solution may rely on the max of the median and current total block commit spend as the denominator and the median commit of the last several blocks as the numerator for calculating the probability that a given miner wins a sortition.
+
+Any ATC solution included in Nakamoto will leave the option for a sortition to have no valid winner. The TenureChange-Extend transaction mitigates the majority of adverse effects caused by a missed sortition.
+
+### Timing of Release from 2.5 to 3.0
+
+Activating Nakamoto will include two epochs:
+
+- **Epoch 2.5:** Pox-4 contract is booted up but no Nakamoto consensus rules take effect.
+- **Epoch 3:** Nakamoto consensus rules take effect.
+
+There will be at least one reward cycle between the initiation of Epoch 2.5 and Epoch 3, but the exact duration will need to be determined prior to activation.
+
+### Updated Block Limits
+
+Block limits in the initial system described in [SIP-001][SIP-001-link] were designed conservatively such that a low compute threshold was required to run a functional node. These cost limits have been updated twice before, as improvements have been made to the stacks-node. This SIP calls for another block limit update to  reflect recent optimizations in the reference Clarity virtual machine.
```

Comment:
I think we want to say something here about the block limits also changing due to the new mechanism in Nakamoto, in which the block creation is no longer a race with Bitcoin.

Reply:
What if we changed the wording of the last sentence to this:

> This SIP calls for another block limit update to reflect recent optimizations in the reference Clarity virtual machine and the new mechanism in Nakamoto in which block creation is no longer a race with Bitcoin.

Reply:
I plan on changing this section to this:

> Block limits in the initial system described in [SIP-001][SIP-001-link] were designed conservatively such that a low compute threshold was required to run a functional node. These cost limits have been updated twice before, as improvements have been made to the stacks-node. This SIP calls for another block limit update to reflect several changes:
> 
> 1. Contract calls are less computationally expensive due to practical optimizations in the Clarity virtual machine.
> 2. Mining a Stacks block is not a race against the next Bitcoin block under Nakamoto.
> 3. The block limit is now spread over multiple Stacks blocks within a miner's tenure.

Reply:
Committed in [358a42c](https://github.com/stacksgov/sips/pull/155/commits/358a42cc188c776fb3559d0a837f708c44abcd2b)